File: typescript/1910.remove-all-occurrences-of-a-substring.ts
/*
 * @lc app=leetcode id=1910 lang=typescript
 *
 * [1910] Remove All Occurrences of a Substring
 */

// @lc code=start
function removeOccurrences(s: string, part: string): string {
    let res = s
    while (res.includes(part)) {
        const pos = res.indexOf(part);
        res = res.slice(0, pos) + res.slice(pos + part.length)
    }
    return res

};
// @lc code=end

File: 1910.remove-all-occurrences-of-a-substring.py
#
# @lc app=leetcode id=1910 lang=python3
#
# [1910] Remove All Occurrences of a Substring
#


# @lc code=start
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        res = s
        while part in res:
            pos = res.index(part)
            res = res[:pos] + res[pos + len(part) :]
        return res


# @lc code=end